python-avd/pyavd/_eos_designs/structured_config/underlay/ethernet_interfaces.py

```diff
@@ -87,17 +82,17 @@ def ethernet_interfaces(self: AvdStructuredConfigUnderlayProtocol) -> None:
                 # MPLS
                 if self.shared_utils.underlay_mpls:
                     ethernet_interface.mpls.ip = True
-                    if self.shared_utils.underlay_ldp is True:
+                    if self.shared_utils.underlay_ldp:
                         ethernet_interface.mpls.ldp._update(interface=True, igp_sync=True)
 
                 # IP address
-                if link.get("ip_address") is not None:
-                    if "unnumbered" in link["ip_address"].lower():
-                        ethernet_interface.ip_address = link["ip_address"]
+                if link.ip_address:
+                    if "unnumbered" in link.ip_address:
```

Comment:
Note to self: do we cover lower in the facts?

Reply:
hmm actually we still use lower below so lets add it here as well to make sure we don't have a breeaking change (or we remove it below)
```suggestion
                    if "unnumbered" in link.ip_address.lower():
```